pkg/auth/authenticator/interfaces.go

```diff
@@ -43,6 +43,11 @@ type Password interface {
 	AuthenticatePassword(user, password string) (user.Info, bool, error)
 }
 
+// Negotiate checks a context initiation token using a GssProxy.
+type Negotiate interface {
```

Comment:
Currently it expects the client's authorization header to be marked as "bearer".  Any preference on how it should be made to accept "negotiate"?